src/main/java/org/jenkinsci/plugins/workflow/graph/StandardGraphLookupView.java

```diff
@@ -25,10 +25,10 @@
     static final String INCOMPLETE = "";
 
     /** Map the blockStartNode to its endNode, to accellerate a range of operations */
-    HashMap<String, String> blockStartToEnd = new HashMap<>();
+    ConcurrentHashMap<String, String> blockStartToEnd = new ConcurrentHashMap<>();
```

Comment:
This should be fine since the value type is just `String` that already exists. Just a reminder that in general you need to be careful about having one thread add an entry to a map (or whatever) that another thread might pull off without the use of any synchronization barriers: if the object were recently constructed, its constructor might not have completed by the time the other thread uses it. One of those weird theoretical problems that probably does not happen often in reality but is hard to reason about.

Reply:
> if the object were recently constructed, its constructor might not have completed by the time the other thread uses it.

Wow, good to know.  Thanks!